### PR TITLE
chore: improve documentation for customer onboarding 2

### PR DESCRIPTION
**Pull request by brianshen3**
### Linked issue(s)
Fixes KOL-7613
### What change does this PR introduce and why?
Added use cases with code examples because clients were requesting for code examples.
### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: docs/dataset/use-cases.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+<div class="grid cards" markdown>
+
+- [:kolena-age-estimation-20: Example: Regression ↗](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/age_estimation)
+
+    ![Example images from the Labeled Faces in the Wild dataset.](../assets/images/LFW.jpg)
+
+    ---
+
+    Age Estimation using the [Labeled Faces in the Wild (LFW)](http://vis-www.cs.umass.edu/lfw/) dataset
```
Comment (nankolena) [inline docs/dataset/use-cases.md]: nit - use `https`

File: mkdocs.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ nav:
     - Quickstart: dataset/quickstart.md
     - Installing <code>kolena</code>: installing-kolena.md
     - Core Concepts: dataset/core-concepts/index.md
+    - Use Cases: dataset/use-cases.md
```
Comment (nankolena) [inline mkdocs.yml:13]: Was thinking if it'd be helpful to include a ref at the bottom of the quick start page to reference this page, maybe in the conclusion section. Any thoughts?

Comment (brianshen3) [reply mkdocs.yml:13]: I started with that. I found the quick start is quite long and the use cases would be hard to find. Maybe I add the use case under quickstart?

File: docs/dataset/use-cases.md (hunk 1 above)
Comment (brianshen3) [reply docs/dataset/use-cases.md]: good catch. Updated `http` to `https`

File: mkdocs.yml (hunk 2 above)
Comment (nankolena) [reply mkdocs.yml:13]: Oh I was thinking just a sentence in the quick start guide, such as check out other examples in the use cases section. Agreed that we don't want to include this whole section in the quick start guide.

Comment (brianshen3) [reply mkdocs.yml:13]: Sorry I misunderstood. Okay I updated to include that. I think its a great idea to have that.

**Review by nankolena — APPROVED** [review]
Nice! cc @munkyshi @mkaramlou in case they have any input

**Pull request merged**